training/distributed_training/pytorch/model_parallel_v2/gpt-neox/smp-train-gpt-neox-fsdp-tp.ipynb

```diff
@@ -662,6 +661,7 @@
     "    \"sharding_strategy\": \"hybrid_shard\",\n",
     "    \"train_batch_size\": 2,\n",
     "    \"use_smp_flash_attn\": 1,\n",
+    "    \"use_smp_implementation\": 1,\n",
```

Comment:
What is this parameter? Do we need to document this at [SMP v2 core feature configuration parameters](https://docs.aws.amazon.com/sagemaker/latest/dg/distributed-model-parallel-v2-reference.html#distributed-model-parallel-v2-reference-init-config)? I also see some more params that look SMP-specific.

Reply:
This parameter is to enable our SMP implementation for supported HF models. These are distinct from the referenced "core feature config params" as those are passed via the sagemaker estimator model parallel parameter swhile these are script-level only, so don't think these should be documented there. Basically, most of these SMP parameters implement the logic defined in the scripts like [TP](https://docs.aws.amazon.com/sagemaker/latest/dg/model-parallel-core-features-v2-tensor-parallelism.html), and the user just has to pass the flag to enable it.